pennylane/gradients/pulse_gradient.py

```diff
@@ -461,10 +532,51 @@ def ansatz(params):
     sampler_seed = sampler_seed or np.random.randint(18421)
     key = jax.random.PRNGKey(sampler_seed)
 
-    return _expval_stoch_pulse_grad(tape, argnum, num_split_times, key, shots)
+    return _expval_stoch_pulse_grad(tape, argnum, num_split_times, key, shots, use_broadcasting)
 
 
-def _expval_stoch_pulse_grad(tape, argnum, num_split_times, key, shots):
+def _generate_tapes_and_cjacs(tape, idx, key, num_split_times, use_broadcasting):
+    """Generate the tapes and compute the classical Jacobians for one given
+    generating Hamiltonian term of one pulse.
+    """
+    op, op_idx, term_idx = tape.get_operation(idx, return_op_index=True)
+    if not isinstance(op, ParametrizedEvolution):
+        raise ValueError(
+            "stoch_pulse_grad does not support differentiating parameters of "
+            "other operations than pulses."
+        )
+    if len(op.t) != 2:
+        raise ValueError(
+            "stoch_pulse_grad does not support differentiating ParametrizedEvolutions with "
+            f"intermediate time steps. Got t={op.t}"
+        )
+
+    coeff, ham = op.H.coeffs_parametrized[term_idx], op.H.ops_parametrized[term_idx]
+    if not qml.pauli.is_pauli_word(ham):
+        raise ValueError(
+            "stoch_pulse_grad currently only supports Pauli words as parametrized "
+            f"terms in ParametrizedHamiltonian. Got {ham}"
+        )
+    word = qml.pauli.pauli_word_to_string(ham)
+    cjac_fn = jax.jacobian(coeff, argnums=0)
+
+    t0, t1 = op.t
+    taus = jnp.sort(jax.random.uniform(key, shape=(num_split_times,)) * (t1 - t0) + t0)
+    cjacs = [cjac_fn(op.data[term_idx], tau) for tau in taus]
+    if use_broadcasting:
+        split_evolve_ops = split_evol_ops(op, word, ham.wires, taus)
+        tapes = _split_evol_tapes(tape, split_evolve_ops, op_idx)
+    else:
+        tapes = []
+        for tau in taus:
```

Comment:
Just to get this right, `split_evol_ops` always broadcasts by default whenever `tau` is not a single float? And that is why here for the case of no broadcasting this is done step by step?

Reply:
Yes, exactly. The boundary between splitting ops and creating tapes is maintained to be in this function.

There is an alternative way of doing this, but I'm not sure it gains us terribly much, and it would restructure `split_evol_ops`, leading to changed tests etc. So I'm not against moving the internal separation around, but don't know whether there is a strong reason _for_ doing so...

Reply:
Is good the way it already is, was just curious! 🙂

Reply:
Alright! :)